arch/arm/src/cmake/Toolchain.cmake

```diff
@@ -188,7 +188,8 @@ set(ARCHCFLAGS "-Wstrict-prototypes")
 if(NOT CONFIG_LIBCXXTOOLCHAIN)
   set(ARCHCXXFLAGS "${ARCHCXXFLAGS} -nostdinc++")
 else()
-  set(ARCHCXXFLAGS "${ARCHCXXFLAGS} -D_STDLIB_H_")
+  set(ARCHCFLAGS "${ARCHCFLAGS} -D__LIBCXXTOOLCHAIN__")
```

Comment:
@leducp why to include `-D__LIBCXXTOOLCHAIN__` to ARCHCFLAGS too? Shouldn't ARCHCXXFLAGS be enough?

Reply:
Because I need to enable some wrapping in the C part (C++ depends in the NuttX libc for the locale).

My goal is to enable a few symbols to be built in the NuttX libc if:
  1.  the config enable the C++ library from the toolchain
  2. the toolchain is GCC